config.py

```diff
@@ -45,14 +51,59 @@ class Settings(BaseSettings):
     OPENAI_API_KEY: SecretStr
     PGPASSWORD: SecretStr = Field(default="password")
     KALEMAT_API_KEY: SecretStr
-    VECTARA_AUTH_TOKEN: SecretStr
-    VECTARA_CUSTOMER_ID: str
-    VECTARA_CORPUS_ID: str
+
+    VECTARA_API_KEY: SecretStr
+
+    MAWSUAH_VECTARA_CORPUS_KEY: str = Field(
+        alias="MAWSUAH_VECTARA_CORPUS_KEY", default="mawsuah_unstructured"
+    )
+    MAWSUAH_FN_NAME: str = Field(default="search_mawsuah")
+    MAWSUAH_FN_DESCRIPTION: str = Field(
+        default="Search and retrieve relevant rulings from the Islamic jurisprudence (fiqh) encyclopedia based on a specific topic. "
+        "Returns a list of potentially relevant matches that may span multiple paragraphs. "
+        "The search will be based on the 'query' parameter, which must be provided."
+    )
+    MAWSUAH_TOOL_PARAMS: list = Field(
+        default=[
+            {
+                "name": "query",
+                "type": "string",
+                "description": "Topic or subject matter to search for within the fiqh encyclopedia. Write the query in Arabic.",
+            }
+        ]
+    )
+    MAWSUAH_TOOL_REQUIRED_PARAMS: list = Field(default=["query"])
+
+    TAFSIR_VECTARA_CORPUS_KEY: str = Field(
+        alias="TAFSIR_VECTARA_CORPUS_KEY", default="tafsirs"
+    )
+    TAFSIR_FN_NAME: str = Field(default="search_tafsir")
+    TAFSIR_FN_DESCRIPTION: str = Field(
+        default="""
+        Queries Tafsir Ibn Kathir (the renowned Qur'anic exegesis) for relevant 
+        interpretations and explanations. You call this function when you need to 
+        provide authoritative Qur'anic commentary and understanding based on Ibn 
+        Kathir's work. Regardless of the language used in the original conversation, 
+        you will translate the query into English before searching the tafsir. The 
+        function returns a list of **potentially** relevant matches, which may include 
+        multiple passages of interpretation and analysis.
```

Comment:
Should this be pulled out into a resource instead of included in the code?

Reply:
Makes total sense since it's part of the prompt.